ESLint 9, flat config migration (#1537)
seek-oss/skuba: .changeset/thick-taxis-vanish.md

---
'eslint-config-skuba': major
---

Migrate to ESLint 9, `@typescript-eslint` 8, `eslint-config-seek` 14.

These changes may affect your project setup if customising your ESLint configuration. See the individual migration guides:

- https://eslint.org/docs/latest/use/migrate-to-9.0.0
- https://typescript-eslint.io/blog/announcing-typescript-eslint-v8

> In addition, through these major upgrades, some lint rules have changed or have been renamed. You will likely need adjust your code after running ESLint.

Reviewer: ```suggestion
Through these major upgrades, some lint rules have changed or have been renamed. You will likely need to adjust your code after running ESLint.
```
